mialab/utilities/pipeline_utilities.py

```diff
@@ -80,15 +84,91 @@ def execute(self) -> structure.BrainImage:
         Returns:
             structure.BrainImage: The image with extracted features.
         """
-        if self.coordinates_feature:
-            atlas_coordinates = fltr_feat.AtlasCoordinates()
-            self.img.feature_images[FeatureImageTypes.ATLAS_COORD] = \
-                atlas_coordinates.execute(self.img.images[structure.BrainImageTypes.T1w])
+        if self.load_features:
+            self._load_features()
+        else:
+            self._compute_features()
 
+        # We can keep it here because it is not computed/loaded/saved.
         if self.intensity_feature:
             self.img.feature_images[FeatureImageTypes.T1w_INTENSITY] = self.img.images[structure.BrainImageTypes.T1w]
             if self.t2_features:
-                self.img.feature_images[FeatureImageTypes.T2w_INTENSITY] = self.img.images[structure.BrainImageTypes.T2w]
+                self.img.feature_images[FeatureImageTypes.T2w_INTENSITY] = self.img.images[
+                    structure.BrainImageTypes.T2w]
+
+        if self.save_features:
+            if self.load_features and not self.overwrite:
+                print(f"If `load_features` == True and `overwrite` == False, save_features should be False!")
+            else:
+                self._save_features()
+
+        self._generate_feature_matrix()
+
+        return self.img
+
+    def _load_features(self):
+        if self.coordinates_feature:
+            self._load_feature(FeatureImageTypes.ATLAS_COORD)
+
+        if self.gradient_intensity_feature:
+            self._load_feature(FeatureImageTypes.T1w_GRADIENT_INTENSITY)
+            if self.t2_features:
+                self._load_feature(FeatureImageTypes.T2w_GRADIENT_INTENSITY)
+
+        if self.texture_contrast_feature:
+            self._load_feature(FeatureImageTypes.T1w_TEXTURE_CONTRAST)
+            if self.t2_features:
+                self._load_feature(FeatureImageTypes.T2w_TEXTURE_CONTRAST)
+
+        if self.texture_dissimilarity_feature:
+            self._load_feature(FeatureImageTypes.T1w_TEXTURE_DISSIMILARITY)
+            if self.t2_features:
+                self._load_feature(FeatureImageTypes.T2w_TEXTURE_DISSIMILARITY)
+
+        if self.texture_correlation_feature:
+            self._load_feature(FeatureImageTypes.T1w_TEXTURE_CORRELATION)
+            if self.t2_features:
+                self._load_feature(FeatureImageTypes.T2w_TEXTURE_CORRELATION)
+
+        if self.edge_feature:
+            self._load_feature(FeatureImageTypes.T1w_EDGES)
+            if self.t2_features:
+                self._load_feature(FeatureImageTypes.T2w_EDGES)
```

Comment:
I guess it would be nice to generalize this long list of ifs :D Maybe as a next step!